src/core/Merger.cpp

```diff
@@ -0,0 +1,520 @@
+        Group* targetChildGroup = context.m_targetRootGroup->findGroupByUuid(sourceChildGroup->uuid());
+        if (!targetChildGroup) {
+            changes
+                << tr("Creating missing %1 [%2]").arg(sourceChildGroup->name()).arg(sourceChildGroup->uuid().toHex());
+            targetChildGroup = sourceChildGroup->clone(Entry::CloneNoFlags, Group::CloneNoFlags);
+            moveGroup(targetChildGroup, context.m_targetGroup);
+            TimeInfo timeinfo = targetChildGroup->timeInfo();
+            timeinfo.setLocationChanged(sourceChildGroup->timeInfo().locationChanged());
+            targetChildGroup->setTimeInfo(timeinfo);
+        } else {
+            bool locationChanged =
+                targetChildGroup->timeInfo().locationChanged() < sourceChildGroup->timeInfo().locationChanged();
+            if (locationChanged && targetChildGroup->parent() != context.m_targetGroup) {
+                changes << tr("Relocating %1 [%2]").arg(sourceChildGroup->name()).arg(sourceChildGroup->uuid().toHex());
+                moveGroup(targetChildGroup, context.m_targetGroup);
+                TimeInfo timeinfo = targetChildGroup->timeInfo();
+                timeinfo.setLocationChanged(sourceChildGroup->timeInfo().locationChanged());
+                targetChildGroup->setTimeInfo(timeinfo);
+            }
+            changes << resolveGroupConflict(context, sourceChildGroup, targetChildGroup);
+        }
+        MergeContext subcontext{context.m_sourceDb,
+                                context.m_targetDb,
+                                context.m_sourceRootGroup,
+                                context.m_targetRootGroup,
+                                sourceChildGroup,
+                                targetChildGroup};
+        changes << mergeGroup(subcontext);
+    }
+    return changes;
+}
+
+Merger::ChangeList
+Merger::resolveGroupConflict(const MergeContext& context, const Group* sourceChildGroup, Group* targetChildGroup)
+{
+    Q_UNUSED(context);
+    ChangeList changes;
+
+    const QDateTime timeExisting = targetChildGroup->timeInfo().lastModificationTime();
+    const QDateTime timeOther = sourceChildGroup->timeInfo().lastModificationTime();
+
+    // only if the other group is newer, update the existing one.
+    if (timeExisting < timeOther) {
+        changes << tr("Overwriting %1 [%2]").arg(sourceChildGroup->name()).arg(sourceChildGroup->uuid().toHex());
+        targetChildGroup->setName(sourceChildGroup->name());
+        targetChildGroup->setNotes(sourceChildGroup->notes());
+        if (sourceChildGroup->iconNumber() == 0) {
+            targetChildGroup->setIcon(sourceChildGroup->iconUuid());
+        } else {
+            targetChildGroup->setIcon(sourceChildGroup->iconNumber());
+        }
+        targetChildGroup->setExpiryTime(sourceChildGroup->timeInfo().expiryTime());
+        // TODO HNH: Since we are updating our own group from the source group, I think we should update the timestamp
+    }
```

Comment:
Is the comment at the right spot (it is a valid question, but I'm confused since this code has nothing to do with the history merge of entries)? In my opinion, we should change the `lastModificationTime` of the `targetChildGroup` since it is updated with data from a remote source.

Reply:
It's for `if (timeExisting < timeOther) {` and this method in general.

Reply:
The issue about the history was discussed at another line of code. But what should be discussed it the TODO. I think, we need to update the modified time of the group the data was merged into to the modified time of the group the data was taken from. Am I right?